pkg/webhook/preflight/nutanix/storagecontainer.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"context"
+	"fmt"
+
+	clustermgmtv4 "github.com/nutanix/ntnx-api-golang-clients/clustermgmt-go-client/v4/models/clustermgmt/v4/config"
+	"k8s.io/utils/ptr"
+
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/external/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
+	carenv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/webhook/preflight"
+)
+
+const (
+	csiParameterKeyStorageContainer = "storageContainer"
+)
+
+type storageContainerCheck struct {
+	nodeSpec *carenv1.NutanixNodeSpec
+	field    string
+	csiSpec  *carenv1.CSIProvider
+	nclient  client
+}
+
+func (c *storageContainerCheck) Name() string {
+	return "NutanixStorageContainer"
+}
+
+func (c *storageContainerCheck) Run(ctx context.Context) preflight.CheckResult {
+	result := preflight.CheckResult{
+		Allowed: true,
+	}
+
+	if c.csiSpec == nil {
+		result.Allowed = false
+		result.Error = true
+		result.Causes = append(result.Causes, preflight.Cause{
+			Message: fmt.Sprintf(
+				"no storage container found for cluster %q",
+				*c.nodeSpec.MachineDetails.Cluster.Name,
+			),
+			Field: c.field,
+		})
+
+		return result
+	}
+
+	if c.csiSpec.StorageClassConfigs == nil {
+		result.Allowed = false
+		result.Causes = append(result.Causes, preflight.Cause{
+			Message: fmt.Sprintf(
+				"no storage class configs found for cluster %q",
+				*c.nodeSpec.MachineDetails.Cluster.Name,
+			),
+			Field: c.field,
+		})
+
+		return result
+	}
+
+	for _, storageClassConfig := range c.csiSpec.StorageClassConfigs {
+		if storageClassConfig.Parameters == nil {
+			continue
+		}
+
+		storageContainer, ok := storageClassConfig.Parameters[csiParameterKeyStorageContainer]
+		if !ok {
+			continue
+		}
+
+		// TODO: check if cluster name is set, if not use uuid.
+		// If neither is set, use the cluster name from the NodeSpec failure domain.
```

Comment:
nit: does this comment belong somewhere in `getCluster` instead?

Reply:
old comment, needs to be removed.

Reply:
Removed in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1136/commits/1aa886798428baf5d7b583d6429f1b503e5d1ea4